portfolio/src/main/java/com/google/sps/testing/fake_servlets/FakeAuthenticationServlet.java

```diff
@@ -0,0 +1,81 @@
+
+package com.google.sps.testing.fake_servlets;
+
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
+import java.lang.Math;
+import com.google.common.base.Strings;
+import com.google.common.collect.Range;
+import com.google.sps.data.EntityProperties;
+import com.google.sps.data.RequestParameters;
+import java.util.stream.Collectors;
+import java.util.HashMap;
+import com.google.sps.configuration.Flags;
+import com.google.sps.testing.fake_data.FakeUserService;
+import com.google.sps.testing.fake_data.FakeUserDatabase;
+
+/** 
+  * Fake servlet that uploads and retrieves user data from fake database. 
+  */
+@WebServlet(Flags.IS_REAL_SERVER ? "/fakeauthenticate" : "/authenticate")
+public class FakeAuthenticationServlet extends HttpServlet {
+
+  @Override
+  public void init() {
+    FakeUserDatabase.put("1", "Sally");
+  }
+
+  /** 
+    * Checks whether the user is currently logged in and provides
+    * a log-out link and username if so, and a log-in link if not. 
+    */
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    FakeUserService userService = FakeUserService.FAKE_USER_SERVICE_INSTANCE;
+    HashMap<String, Object> authenticationStatusInfo = new HashMap<String, Object>();
+
+    if (userService.isUserLoggedIn()) {
+      authenticationStatusInfo.put("isLoggedIn", true);
+      String logOutLink = userService.createLogoutURL("/index.html");
+      authenticationStatusInfo.put("logOutLink", logOutLink);
+      authenticationStatusInfo.put("username", getUserName(
+        userService.getUserId()));
+    } else {
+      authenticationStatusInfo.put("isLoggedIn", false);
```

Comment:
Yes, I agree. I had originally put in this much functionality because I thought I might want to eventually flesh out the testing to including logging in/out, but I don't think that'll happen, so I'll fix this.